### ISSUE TITLE
Scaffolding dual assembly

### ISSUE THREAD

**Issue by Tierhon** [chhylp123/hifiasm#216]
Hi,
now that hifiasm 'generates a pair of haplotype-resolved assemblies', what is the best way of scaffolding these? This has been on my mind a bit lately, and this might not be the best place to ask, but I'm certain that many of the knowledgeable people that develop and use hifiasm have some input. 

If you take both assemblies, merge them together in a fasta file and then map HiC reads to it, would that be a good enough signal for many scaffolders? Would many of the read pairs map multiple places and therefore have low Q and be filtered out, even though they could have been informative? A quick test with SALSA lead to quite poor scaffolding, with the N50 of scaffolds only about 15 % as large as when scaffolding the primary contigs.

Further, if you do scaffold each haplotype-resolved assembly on it's own, could that lead to errors? For many species there are likely smaller or larger rearrangements between the two homologous chromosomes, whereby mapping all HiC data to each assembly could confuse that signal. 

Are there a way to split the HiC reads into two piles, one for each haplotype? One could run something like DipAsm, but that seems a bit involved and messy. 

Any answers and pointers are much appreciated. 

Thank you.

Sincerely,
Ole

**Comment by m-jahani** [comment]
That is my question too, please let me know if you find any answer for it.
Thanks,
Mojtaba

**Comment by Tierhon** [comment]
What I have been doing so far is using all HiC data to scaffold each assembly/haplotype by itself. If there are large differences between the haplotypes, the resulting scaffolding might not be optimal, but I cannot really think of any other way of doing this with the tools I am aware of. So I take hap1 and scaffold with SALSA, Juicer, yahs, pins, or whatever, and do the same with hap2 with the same data.

Ole

**Comment by m-jahani** [comment]
Yes, I am doing the same thing. However, the results are not promising so far. There are large differences between the two haplotypes.

**Comment by chhylp123** [comment]
Probably you could try this solution: https://github.com/baozg/phased-assembly-check. We have not run it by ourselves but it is reasonable.

**Comment by Tierhon** [comment]
Doesn't that require trio data? We don't have that for our species. I guess
it is not straightforward doing similar with just HiFi + HiC.

tor. 24. feb. 2022, 20:32 skrev chhylp123 ***@***.***>:

**Comment by chhylp123** [comment]
I guess Hi-C only could work. The idea is to use Hi-C data to identify which contigs are misassigned to wrong haplotypes.

**Comment by paulhavlak** [comment]
I'm wondering about this, too.
The question is how to get only the Hi-C reads you want for that haplotype, without all the noise of other reads.
What I plan to try is as follows:
* Align all Hi-C read-pairs to the full assembly of contigs/scaffolds from all haplotypes. Keep only those read-pairs with at least one end having nonzero MapQ (assuming your aligner gives nonzero MapQ for at most one location per read).
* If both ends have nonzero MapQ, keep only those consistent with the Hifiasm's phasing, discarding others, or else use the phase-conflicted pairs to make corrections.
* Now separate the phases for re-alignment, aligning read-pairs only to the phase where at least one end aligned in the first pass. This should rescue mates that had no alignment with positive MapQ before, if that was because of a contig cloned between phases (e.g. because of identity by descent). At the expense of ideally a tolerable amount of noise. (The minimum MapQ required on the first pass will be one knob worth tuning here.)
* Scaffold each phase separately, using only its custom subset of Hi-C read-pairs with new MapQ scores from the second-pass re-alignment and disjoint from those of other phase(s).

One reason I haven't tried this myself yet is that my challenging genomes are auto-tetraploid, so read four haplotypes instead of two phases into the above. Still many interesting subtleties to resolve.

**Comment by jennyp76** [comment]
@olekto @chhylp123 @m-jahani @paul-havlak-driscolls 

I would love to know if anyone has found any effective solutions or workarounds for this issue. 
Any advice you could share would be of great help.
